### ISSUE TITLE
How to mark an invoice as completed?

### ISSUE THREAD

**Issue by SteveEdson** [bytefury/crater#467]
Back when I was using V3.1 I could mark an invoice as completed. Now, even if I record a payment for the full amount of the invoice, it still shows as due. I can't see anywhere in the UI to change this.

Is this a bug? I'm on 4.1.0

Thanks

**Comment by mwikala** [comment]
Hi @SteveEdson you can definitely still mark invoices as completed. You're correct, the manual option is gone and an invoice is classed as complete when it has a recorded payment for the full amount.

Is there anything in your error logs that would suggest something is wrong? Sharing that on this issue will help us to dig deeper.

**Comment by SteveEdson** [comment]
I'll take a look when I'm next to my machine again, I'm pretty sure I recorded the postmen's using the Android mobile app if that could make a difference though.

**Comment by mwikala** [comment]
Let me know. I tested this on Android and recording a payment automatically marked the invoice as complete.

**Comment by SteveEdson** [comment]
I'm not getting any errors in the logs but I'm able to replicate the issue each time using the android app, it seems to work fine using the web version.

- Create an invoice on the web version (guessing on the mobile app too)
  - (Selected existing customer, selected existing line item)
- Mark invoice as sent on mobile app
- Record payment on mobile app, don't change any of the default values, payment is for the full amount
- Save the payment, invoice shows as 0 outstanding, but still 'due'

**Comment by SteveEdson** [comment]
Also, since the 'completed' status isn't computed just on outstanding == 0, can we have the option back to set it manually please? This might be useful if an invoice is raised, but later on an agreement is reached where the client doesn't pay. The invoice is not longer due, but there's no payment to record, and I wouldn't want it to be deleted from the records.

**Comment by SteveEdson** [comment]
I've figured out the issue, it's not a code issue, its a UX issue on the Android app:

- Open the invoice on the android app, select record payment from the 3 dots menu at the top right
- Record the payment for the full amount and save the payment
- (On the web app, you can see at this point that the invoice is `complete`)
- The app takes you back to the invoice detail screen, so naturally I clicked save (thinking I was updating that invoice / generally just clicking save to go back to the previous screen)
- The invoice is now due again, even though there's no status field on the invoice detail screen, I'm guessing its still part of the laravel fillable attributes and its being overwritten and reverted back to the due state

**Comment by mwikala** [comment]
Good catch! It would have took me a while to replicate this. Do you mind raising this issue on the [mobile app repo](https://github.com/bytefury/crater-mobile)? Then the mobile devs can take a look.
